flyteidl/protos/flyteidl/admin/agent.proto

```diff
@@ -53,18 +58,46 @@ message CreateTaskRequest {
 
 // Represents a create response structure.
 message CreateTaskResponse {
+  // ResourceMeta is created by the agent. It could be a string (jobId) or a dict (more complex metadata).
+  bytes resource_meta = 1;
+}
+
+message CreateRequestHeader {
+  // Template of the task that encapsulates all the metadata of the task.
+  core.TaskTemplate template = 1;
+  // Prefix for where task output data will be written. (e.g. s3://my-bucket/randomstring)
+  string output_prefix = 2;
+  // subset of runtime task execution metadata.
+  TaskExecutionMetadata task_execution_metadata = 3;
+  // MaxDatasetSizeBytes is the maximum size of the dataset that can be generated by the task.
+  int64 max_dataset_size_bytes = 4;
+}
+
+
+message ExecuteTaskSyncRequest {
+  oneof part {
+    CreateRequestHeader header = 1;
+    core.LiteralMap inputs = 2;
+  }
+}
+
+message ExecuteTaskSyncResponseHeader {
+  Resource resource = 1;
+}
+
+message ExecuteTaskSyncResponse {
   // Metadata is created by the agent. It could be a string (jobId) or a dict (more complex metadata).
   // Resource is for synchronous task execution.
   oneof res {
-    bytes resource_meta = 1;
-    Resource resource = 2;
+    ExecuteTaskSyncResponseHeader header = 1;
+    core.LiteralMap outputs = 2;
   }
 }
 
 // A message used to fetch a job resource from flyte agent server.
 message GetTaskRequest {
   // A predefined yet extensible Task type identifier.
-  string task_type = 1;
+  TaskType task_type = 1;
```

Comment:
cc @EngHabu Shall we ensure that this is backward compatible? maybe change it to
```protobuf
message GetTaskRequest {
    string task_type = 1;
     int32 task_type_version = 2;
     ...
```

Reply:
How about we deprecate the old one and create a new field?

Reply:
Just because this is used in many places and consolidation is good here...